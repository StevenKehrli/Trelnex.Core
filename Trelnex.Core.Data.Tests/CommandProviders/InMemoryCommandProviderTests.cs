namespace Trelnex.Core.Data.Tests.CommandProviders;

public class InMemoryCommandProviderTests : CommandProviderTests
{
    [SetUp]
    public async Task TestFixtureSetup()
    {
        // create our command provider
        var factory = await InMemoryCommandProviderFactory.Create();

        _commandProvider =
            factory.Create<ITestItem, TestItem>(
                typeName: "test-item",
                TestItem.Validator,
                CommandOperations.All);
    }
}
