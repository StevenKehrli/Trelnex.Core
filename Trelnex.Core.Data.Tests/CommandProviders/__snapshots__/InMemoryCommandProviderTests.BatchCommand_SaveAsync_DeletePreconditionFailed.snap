{
  "saved": [
    {
      "HttpStatusCode": "FailedDependency",
      "ReadResult": null
    },
    {
      "HttpStatusCode": "Conflict",
      "ReadResult": null
    }
  ],
  "read": [
    {
      "Item": {
        "Id": "0ffc3bd2-b33f-4902-a4bf-b804d8aa01e9",
        "PartitionKey": "c25361d6-3d14-452d-a99d-22ef7c85d70e",
        "TypeName": "test-item",
        "CreatedDate": "2025-01-10T16:17:48.896004Z",
        "UpdatedDate": "2025-01-10T16:17:48.896004Z",
        "DeletedDate": null,
        "IsDeleted": null,
        "ETag": "3d892e77-83da-4b34-b741-15d5cd285512",
        "Message": "Message #1"
      }
    }
  ]
}
